src/features/public/PublicUserProfile/components/ProjectSnippet.tsx

```diff
@@ -12,79 +19,136 @@ interface Props {
 }
 
 export default function ProjectSnippet({ project, key }: Props): ReactElement {
+  const router = useRouter();
   const { t, i18n } = useTranslation(['donate', 'common', 'country']);
-
+  const { theme } = React.useContext(ThemeContext);
+  const [open, setOpen] = React.useState(false);
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const handleOpen = () => {
+    setOpen(true);
+  };
   const ImageSource = project.image
     ? getImageUrl('project', 'medium', project.image)
     : '';
   const progressPercentage =
     (project.countPlanted / project.countTarget) * 100 + '%';
 
-  const projectDetails = project;
+  const [
+    taxDeductionCountriesArray,
+    setTaxDeductionCountriesArray,
+  ] = React.useState([]);
+
+  React.useEffect(() => {
+    let taxarray = [];
+    for (let i = 0; i < project.paymentSetup.taxDeduction.length; i++) {
+      const countryCode = getCountryDataBy(
+        'countryName',
+        project.paymentSetup.taxDeduction[i]
+      ).countryCode;
+      taxarray.push(countryCode);
+    }
+    setTaxDeductionCountriesArray(taxarray);
+  }, []);
+  const projectDetails = {
+    id: project.id,
+    name: project.name,
+    currency: project.currency,
+    country: project.country,
+    treeCost: project.treeCost,
+    taxDeductionCountries: taxDeductionCountriesArray,
+    tpo: project.tpoData,
+  };
```

Comment:
projectDetails meanwhile are filled with `project.properties` as returned by https://app-development.plant-for-the-planet.org/app/projects?_scope=map&currency=EUR - this data structure does not work anymore, so we will have change this structure here for the DonationsPopup component to work:

<img width="716" alt="Bildschirmfoto 2020-10-29 um 18 10 54" src="https://user-images.githubusercontent.com/1532418/97608068-6c2e3d00-1a44-11eb-9701-505544f1c927.png">

Reply:
At best we would be able to load these project properties for a single project before opening the donation popup? Is there an API call for this?

Reply:
I fixed it by doing single API calls for every project. Working but at the price of several API calls.